### ISSUE TITLE
Update request: ouch 0.3.1 → 0.4.1 (backport from unstable)

### ISSUE THREAD

**Issue by tshipenchko** [NixOS/nixpkgs#227555]
- Package name: ouch
- Latest released version: 0.4.1
- Current version on the unstable channel: 0.4.1
- Current version on the stable/release channel: 0.3.1
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

It was about 2 year from latest backport

Notify maintainers
@figsoda @psibi

-----

Note for maintainers: Please tag this issue in your PR.

**Comment by figsoda** [comment]
ouch 0.4.1 uses feature from rust 1.65, which isn't available in Nixpkgs/NixOS 22.11. Compiling ouch 0.4.1 fails with this error:

```rust
 error[E0658]: `let...else` statements are unstable
    --> src/extension.rs:147:5
     |
 147 | /     let Some(mut name) = path.file_name().and_then(<[u8] as ByteSlice>::from_os_str) else {
 148 | |         return (path, extensions);
 149 | |     };
     | |______^
     |
     = note: see issue #87335 <https://github.com/rust-lang/rust/issues/87335> for more information
```

NixOS 23.05 will release in a little over a month, which will get ouch 0.4.1, hopefully that would help. You can track the schedule in https://github.com/NixOS/nixpkgs/issues/223562

**Comment by tshipenchko** [comment]
Thanks for your response! I will wait next release then
